src/PeopleSDKAdapter.js

```diff
@@ -62,13 +65,13 @@ export default class PeopleSDKAdapter extends PeopleAdapter {
    * @memberof PeopleSDKAdapter
    */
   getMe() {
-    return from(this.fetchPerson('me')).pipe(
+    return defer(() => this.fetchPerson('me')).pipe(
```

Comment:
What benefit does using `defer` here give us? Could you add some comments around why things are done this way? I'm sure other contributors will look to these functions as reference for new ones being created and it will help to understand the reasoning since the rxjs stuff is still brand new to most of us.

Reply:
form what I learned, `defer` triggers the action only if there is a subscription. this makes the code a bit more performant. In our case, `defer` resolves the promise only if there is a subscription.  **However**, I don't know this would be much beneficial in our case. where we always subscribe to the function that returns an observable. Thats means `defer` does what `from` does essentially. resolving the promise right after invoking the parent function.

Reply:
There is a good explanation on this [StackOverflow answer on `defer` vs `from`](https://stackoverflow.com/questions/39319279/convert-promise-to-observable/54111045#54111045).

In summary, the main difference between `defer` and `from` is **when** the promise gets executed. With `from`, the promise gets executed right away, even when the observable is not subscribed to (this is one of the reasons that we would get "weird" executions flows when debugging), and because of that, the promise only gets resolved once. This means that even when we subscribe later in the code to the same `from` observable, we would still get old values because it will returned whatever the promised had resolved to previously.
With `defer`, the promise gets executed on subscription, so every new subscription to `defer` will execute the promise again, returning a fresh value. So using `defer` will allow us to always get fresh data in-between subscriptions, and it also works more inline with the mental flow of the code

Reply:
shouldn't this change be applied to all of our adapters then? perhaps a different PR/JIRA? I don't personally think we should have it here, as this is more `perf` work than a `fix`

Reply:
Sure, it can but I don't think the work needs to be removed just because it's not everywhere else, if we know this is the right thing to do

Reply:
to me this is still a `perf` not a `fix`

Reply:
It is a `fix` because it is fixing the error handling when a user has presence turned off. When changing `defer` in other PRs it can be a `perf`

Reply:
`defer` is not a `fix`.

Reply:
Sounds good, thanks for the explanation! We've got lots of comments in the `getPerson` function, could you add the same here?

Reply:
Add why using `defer`? I'm not sure that's necessary. It's a pattern for converting promises => Observables, and we didn't comment on the usage of `from`, because of the same